app/models/concerns/archivable.rb

```diff
@@ -0,0 +1,234 @@
+    'created_by_id'               => :save,
+    'updated_at'                  => :save,
+    'version'                     => :save,
+    'appointment_date'            => :save,
+    'race_ethnicity'              => :save,
+    'city'                        => :save,
+    'state'                       => :save,
+    'county'                      => :save,
+    'employment_status'           => :save,
+    'income'                      => :save,
+    'insurance'                   => :save,
+    'referred_to_clinic'          => :save,
+    'resolved_without_fund'       => :save,
+    'clinic_id'                   => :save,
+    'urgent_flag'                 => :save,
+    'procedure_cost'              => :save,
+    'patient_contribution'        => :save,
+    'naf_pledge'                  => :save,
+    'fund_pledge'                 => :save,
+    'pledge_sent'                 => :save,
+    'pledge_generated_at'         => :save,
+    'pledge_generated_by_id'      => :save,
+  }.freeze
+
+  FULFILLMENT_ARCHIVE_REFERENCE = {
+    :check_number           => :shred,
+    :_id                    => :save,
+    :created_by_id          => :save,
+    :updated_at             => :save,
+    :created_at             => :save,
+    :version                => :save,
+    :fulfilled              => :save,
+    :procedure_date         => :save,
+    :gestation_at_procedure => :save,
+    :procedure_cost         => :save,
+    :date_of_check          => :save,
+  }.freeze
+
+  NOTES_ARCHIVE_REFERENCE = {
+    :full_text     => :shred,
+    :_id           => :save,
+    :created_at    => :save,
+    :created_by_id => :save,
+    :updated_at    => :save,
+    :version       => :save,
+  }.freeze
+
+  PLEDGES_ARCHIVE_REFERENCE = {
+    :_id           => :save,
+    :active        => :save,
+    :source        => :save,
+    :amount        => :save,
+    :created_at    => :save,
+    :created_by_id => :save,
+    :updated_at    => :save,
+    :version       => :save,
+  }.freeze
+
+
+  def has_alt_contact?
+    other_contact.present? || other_phone.present? || other_contact_relationship.present?
+  end
+
+  def age_range
+    case age
+    when nil, ''
+      nil
+    when 1..17
+      'Under 18'
+    when 18..24
+      '18-24'
+    when 25..34
+      '25-34'
+    when 35..44
+      '35-44'
+    when 45..54
+      '45-54'
+    when 55..100
+      '55+'
+    else
+      'Bad value'
+    end
+  end
+
+  def save
+    # field can be returned as is
+    # no action necesary, just avoids flagging to delete
+  end
+
+  def shred(field)
+    self[field.to_sym] = nil
+  end
+
+  def scramble(field)
+    # TODO field should be one-way hashed (with salt)
+  end
+
+  def convert_age
+  end
+
+  # TODO
+  def archive_fulfillment
+  end
+
+  # TODO
+  def archive_calls
+  end
+
+  # TODO
+  def archive_notes
+  end
+
+  # TODO
+  def archive_pledges
+  end
+
+
+  def archive!
+    self.archived = true
+    self.had_other_contact = self.has_alt_contact?
+
```

Comment:
@colinxfleming could I get your eyes on this function? If I invoke the shred function directly/hackily (like lines 176-177), the patient object is properly updated to have nil in those fields. However if I invoke the shred function inside of the attributes loop here, nothing is changed. 

I'm testing in the console to get these responses. This current version is also simplified (see line 160 for the original `if`, currently replaced with the `if`+`elsif` to eliminate complications)

I suspect I'm tripping on ruby or rails syntax here. Any ideas?